node/src/chain/acceptor.rs

```diff
@@ -160,9 +160,21 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network> Acceptor<N, DB, VM> {
     }
 
     fn needs_update(blk: &Block, txs: &[SpentTransaction]) -> bool {
+        // Update provisioners at every epoch (where new stakes take effect)
         if blk.header().height % EPOCH == 0 {
             return true;
         }
+        // Update provisioners if a slash has been applied
+        if blk
+            .header()
+            .failed_iterations
+            .cert_list
+            .iter()
+            .any(|i| i.is_some())
```

Comment:
Maybe we should `debug` log here the slashed public keys for debugging. We can also `warn` log, if my public key is amongst the slash list.

Reply:
I added an INFO in the accept_block